apps/server/src/adapters/WebsocketAdapter.ts

```diff
@@ -103,104 +104,57 @@ class SocketServer implements IAdapter {
       ws.on('message', (data) => {
         try {
           // @ts-expect-error -- this works fine
-          const message = JSON.parse(data);
+          const message = JSON.parse(data) as WebSocketPacketToServer;
           const { type, payload } = message;
 
-          if (type === 'ping') {
-            ws.send(
-              JSON.stringify({
-                type: 'pong',
-                payload,
-              }),
-            );
-            return;
-          }
-
-          if (type === 'get-client-name') {
-            ws.send(
-              JSON.stringify({
-                type: 'client-name',
-                payload: this.getOrCreateClient(clientId),
-              }),
-            );
-            return;
-          }
-
-          if (type === 'set-client-patch') {
-            if (payload && typeof payload == 'object') {
-              this.clients.set(clientId, { ...this.clients.get(clientId), ...payload });
+          switch (type) {
+            case WsType.PING: {
+              sendPacket({ type: WsType.PONG, payload });
+              break;
             }
-            this.sendClientList();
-            return;
-          }
-
-          if (type === 'set-client-type') {
-            if (payload && typeof payload == 'string') {
+            case WsType.CLIENT_SET: {
               const previousData = this.getOrCreateClient(clientId);
-              this.clients.set(clientId, { ...previousData, type: payload });
+              this.clients.set(clientId, { ...previousData, ...payload });
```

Comment:
question: why do we need to spread payload now?
it is unclear in the typing which may cause issues in the future, maybe it warrants a comment?

Reply:
it is partial data